src/types/timeout-error.ts

```diff
@@ -0,0 +1,5 @@
+export class TimeoutError extends Error {
```

Comment:
Same question as the previous error comments. Why not just leave this as `Error` and pass in a message stating that its a `TimeoutError`?